packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +17,75 @@ declare global {
 /**
  * A nimble styled rich text viewer
  */
-export class RichTextViewer extends FoundationElement {}
+export class RichTextViewer extends FoundationElement {
+    /**
+     * @public
+     * @remarks
+     * Accessor: uses the parsing logic to convert the input markdown string to render in the component.
+     */
+    public set markdownValue(value: string) {
+        this._markdownValue = value;
+        this.serializedContent = this.parseMarkdownToDOM(value);
+    }
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: gets the raw markdown string.
+     */
+    public get markdownValue(): string {
+        return this._markdownValue;
+    }
+
+    public serializedContent?: HTMLElement | DocumentFragment;
+    private _markdownValue = '';
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.appendSerializedContentToViewer();
+    }
+
+    /**
+     *
+     * This function takes a markdown string, parses it using the ProseMirror MarkdownParser, serializes the parsed content into a
+     * DOM structure using a DOMSerializer, and returns the serialized result.
+     *
+     * @internal
+     */
+    public parseMarkdownToDOM(value: string): HTMLElement | DocumentFragment {
+        const serializer = DOMSerializer.fromSchema(schema);
+
+        /**
+         * It configures the tokenizer of the default Markdown parser with the 'zero' preset.
+         * The 'zero' preset is a configuration with no rules enabled by default to selectively enable specific rules.
+         * https://github.com/markdown-it/markdown-it/blob/master/lib/presets/zero.js#L1
+         *
+         */
+        const zeroTokenizerConfiguration = defaultMarkdownParser.tokenizer.configure('zero');
+        const supportedTokenizerRules = zeroTokenizerConfiguration.enable([
+            'emphasis',
+            'list',
+            'autolink'
+        ]);
+
+        const parser = new MarkdownParser(
+            schema,
+            supportedTokenizerRules,
+            defaultMarkdownParser.tokens
+        );
```

Comment:
@vikisekarNI, Thanks for pointing this out. Yes, it is better to move the initialization of `DOMSerializer` and `MarkdownParser` outside this function. I have moved it to a `constructor()` for the initialization. 

As I mentioned in the previous comment, the initial though is to reuse the same function for `nimble-rich-text-editor` to parse the markdown input to load it in the editor. With this approach, it will encapsulate all the initialization that is required for parsing markdown input to render it in the component. So we can directly call this function from the editor. 

But it is better not to have a single function to communicate between components. So made changes as per your suggestion.